route/v1/types.go

```diff
@@ -225,6 +229,44 @@ type RouteIngressCondition struct {
 	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,5,opt,name=lastTransitionTime"`
 }
 
+// RouteCertificateConditionType is a valid value forRouteCertificateCondition
+type RouteCertificateConditionType string
+
+// RouteCertificateCondition contains details of the certificate being served by routers associated with
+// this route.
+type RouteCertificateCondition struct {
```

Comment:
Apart from type checking on the `Type` field within this, is there any reason not to just use a `metav1.Condition`?